### PR TITLE
[zh-cn]: Update translation of String.codePointAt()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13975

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md [hunk 1]
```diff
@@ -105,6 +88,7 @@ if (!String.prototype.codePointAt) {
 
 ## 相关链接
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md:89]:
```suggestion
## 参见
```

File: files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md [hunk 2]
```diff
@@ -5,94 +5,77 @@ slug: Web/JavaScript/Reference/Global_Objects/String/codePointAt
...
+```js-nolint
+codePointAt(index)
 ```
 
 ### 参数
 
-- `pos`
-  - : 这个字符串中需要转码的元素的位置。
+- `index`
+  - : 需要返回的字符的基于零的索引。会将其[转换为整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#整数转换)——`undefined` 会转换为 0。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md:21]:
```suggestion
  - : 需要返回的字符的（从零开始的）索引。会被[转换为整数](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number#整数转换)——`undefined` 会转换为 0。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md [hunk 3]
```diff
@@ -5,94 +5,77 @@ slug: Web/JavaScript/Reference/Global_Objects/String/codePointAt
...
 
 ### 返回值
 
-返回值是在字符串中的给定索引的编码单元体现的数字，如果在索引处没找到元素则返回 {{jsxref("undefined")}} 。
+一个非负整数，表示给定 `index` 处字符的码位值。
+
+- 如果 `index` 超出了范围 `0` - `str.length - 1`，`codePointAt()` 返回 {{jsxref("undefined")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md:27]:
```suggestion
- 如果 `index` 超出了 `0` – `str.length - 1` 的范围，`codePointAt()` 返回 {{jsxref("undefined")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md [hunk 4]
```diff
@@ -5,94 +5,77 @@ slug: Web/JavaScript/Reference/Global_Objects/String/codePointAt
...
+- 如果 `index` 处的元素是一个 UTF-16 前导代理（leading surrogate），则返回代理*对*的码位。
+- 如果 `index` 处的元素是一个 UTF-16 后尾代理（trailing surrogate），则*只*返回后尾代理的码元。
 
 ## 描述
 
-如果在指定的位置没有元素则返回 {{jsxref("undefined")}}。如果在索引处开始没有 UTF-16 代理对，将直接返回在那个索引处的编码单元。
+字符串中的字符从左到右进行索引。第一个字符的索引为 `0`，而字符串 `str` 中最后一个字符的索引为 `str.length - 1`。
 
-Surrogate Pair 是 UTF-16 中用于扩展字符而使用的编码方式，是一种采用四个字节 (两个 UTF-16 编码) 来表示一个字符，称作代理对。
+Unicode 码位范围从 `0` 到 `1114111`（`0x10FFFF`）。在 UTF-16 中，每个字符串索引是一个取值范围为 `0` - `65535` 的码元。较高的码位由一个由*一对* 16 位代理伪字符表示。因此，`codePointAt()` 返回的码位可能跨越两个字符串索引。有关 Unicode 的信息，请参阅[UTF-16 字符、Unicode 码位和字素簇](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#utf-16_字符、unicode_码位和字素簇)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md:35]:
```suggestion
Unicode 码位范围从 `0` 到 `1114111`（`0x10FFFF`）。在 UTF-16 中，每个字符串索引是一个取值范围为 `0` – `65535` 的码元。较高的码位由一个由*一对* 16 位代理伪字符表示。因此，`codePointAt()` 返回的码位可能跨越两个字符串索引。有关 Unicode 的信息，请参阅 [UTF-16 字符、Unicode 码位和字素簇](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#utf-16_字符、unicode_码位和字素簇)。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md [hunk 5]
```diff
@@ -5,94 +5,77 @@ slug: Web/JavaScript/Reference/Global_Objects/String/codePointAt
...
+"\ud83d\ude0d".codePointAt(1); // 56845
+"\ud83d\ude0d".codePointAt(1).toString(16); // de0d
+
+"ABC".codePointAt(42); // undefined
 ```
 
-## 替补支持（Polyfill）
+### 在循环中使用 codePointAt()
+
+因为使用字符串索引进行循环会导致同一个代码点被访问两次（一次是前导代理，一次是后尾代理），而第二次调用 `codePointAt()` 时*只*返回后尾代理项，所以最好避免使用索引进行循环。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md:58]: align the context

```suggestion
因为使用字符串索引进行循环会导致同一码位被访问两次（一次是前导代理，一次是后尾代理），而第二次调用 `codePointAt()` 时*只*返回后尾代理项，所以最好避免使用索引进行循环。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md [hunk 6]
```diff
@@ -5,94 +5,77 @@ slug: Web/JavaScript/Reference/Global_Objects/String/codePointAt
...
 
-给原生不支持 ECMAScript 6 的浏览器使用`codePointAt()`方法的的一个字符串扩展方法。
+```js example-bad
+const str = "\ud83d\udc0e\ud83d\udc71\u2764";
+
+for (let i = 0; i < str.length; i++) {
+  console.log(str.codePointAt(i).toString(16));
+}
+// '1f40e', 'dc0e', '1f471', 'dc71', '2764'
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md:66]:
```suggestion
// '1f40e'、'dc0e'、'1f471'、'dc71'、'2764'
```

File: files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md [hunk 7]
```diff
@@ -5,94 +5,77 @@ slug: Web/JavaScript/Reference/Global_Objects/String/codePointAt
...
-        writable: true,
-      });
-    } else {
-      String.prototype.codePointAt = codePointAt;
-    }
-  })();
+for (const codePoint of str) {
+  console.log(codePoint.codePointAt(0).toString(16));
 }
+// '1f40e', '1f471', '2764'
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/codepointat/index.md:75]:
```suggestion
// '1f40e'、'1f471'、'2764'
```

**Review by yin1999 — COMMENTED** [review]
Almost looks good to me. Just some nits :)

**Review by yin1999 — APPROVED** [review]
Thanks @JasonLamv-t :tada:

**Pull request merged**